src/remoteConnector.ts

```diff
@@ -490,7 +492,37 @@ export default class RemoteConnector extends Disposable {
 			this.logger.error(`Couldn't write '${sshDestInfo.hostName}' host to known_hosts file`, e);
 		}
 
-		return Buffer.from(JSON.stringify(sshDestInfo), 'utf8').toString('hex');
+		// Connect to the OpenSSH agent and check for registered keys
```

Comment:
@iQQBot could you comment on how reliable it is?

Reply:
This is the same thing SSH does internally it connects to the agent and get the public keys

Reply:
I have mine started in macOS, don't remember having to change something to start it automatically 🤔
I guess without the agent running you still can connect using ssh gateway right @iQQBot ? Could you update your settings with `"remote.SSH.logLevel": "trace"` and share the remote-ssh extension logs

Reply:
vscode docs says agent run by default on mac https://code.visualstudio.com/docs/remote/troubleshooting#_setting-up-the-ssh-agent

Reply:
I checked again and I will test for the key files manually not using the agent